### PR TITLE
VTag improvements

### LINKED ISSUE #3190: Refactor and improve `VTag` component
## Description

The `VTag` component added in #3137 can be improved further in the following ways:

- forward all props to the inner `VButton` (and potentially also events from the inner `VButton`?).
- use a slot for the content instead of a prop. This can unlock future UI additions/improvements such as icons to indicate ML-generated tags etc.
- add accessible labels to the links to clearly indicate that the link is a tag.

## Additional context

This issue was based on a review comment: https://github.com/WordPress/openverse/pull/3137#pullrequestreview-1675188420

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #3190 by @dhruvkb

This PR is the updated version of #3870 by @Stagge

## Description

The VTag component added in https://github.com/WordPress/openverse/pull/3137 has been improved in the following ways:

- All props are forwarded to the inner `VButton` (and the `VLink` component)
- All events are emitted from the inner `VButton`
- The content is displayed with a slot for the content instead of a prop.
- ~~Accessible labels to the links have been added to clearly indicate that the link is a tag.~~ The a11y group advised not to add aria-labels, and use the inner text instead
- Tests for the VTag component has been added.

## Testing Instructions

Run the frontend app using `just frontend/run dev`
Go to `http://localhost:8443/preferences` and switch the `additional_search_views` flag on.
Go to the homepage and search for something, then select one of the results. You should see the tags work as expected.

There are no visual changes, so the VR tests pass.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/test/unit/specs/components/VTag/v-tag.spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+describe("VTag", () => {
+  let props = null
+  let options = null
+
+  beforeEach(() => {
+    props = {}
+    options = { propsData: props }
+  })
+
+  it("should render an anchor tag by default", () => {
```
Comment (zackkrida) [inline frontend/test/unit/specs/components/VTag/v-tag.spec.js:16]: Something like this might be more useful, i haven't tested though:

```ts
options.slots = {
  default: "Hello world",
}

const link = screen.getByRole('link', { name: "Hello world" })
expect(link).toHaveAttribute('href', 'https://example.com/')
```

This seems like the primary case we want to test, that a link is rendered with the correct text and href.

**Review by zackkrida — APPROVED** [review]
I think the tests could use some cleanup but that it isn't blocking.

File: frontend/src/components/VTag/VTag.vue [hunk 2]
```diff
@@ -4,9 +4,10 @@
     size="small"
     variant="filled-gray"
     class="label-bold"
-    :href="href"
-    >{{ title }}</VButton
-  >
+    v-bind="$props"
+    v-on="$listeners"
```
Comment (dhruvkb) [inline frontend/src/components/VTag/VTag.vue:8]: Why were these added?

**Review by dhruvkb — APPROVED** [review]
LGTM, I just have a doubt.

File: frontend/src/components/VTag/VTag.vue (hunk 2 above)
Comment (obulat) [reply frontend/src/components/VTag/VTag.vue:8]: Normally, these should just pass through, but I think because we use 2 levels of components here: `VButton` that is `as` `VLink`, they are not being passed without these.
